packages/data-broker/src/types/models.ts

```diff
@@ -94,52 +93,65 @@ export type DataServiceEntity = {
   entity_code: string;
 };
 
-export type DataServiceSyncGroup = {
-  code: string;
-  service_type: ServiceType;
-  config: Record<string, any>;
-  data_group_code: string;
-  sync_cursor: string;
-  sync_status: SYNC_STATUS;
-  databaseType: typeof TYPES.DATA_SERVICE_SYNC_GROUP;
-};
-
 export type Entity = {
   code: string;
   name: string;
+  type: string;
   point?: string;
   config: {
     kobo_id?: string;
   };
-  country_code?: string;
-  type: string;
+  metadata: {
+    dhis?: {
+      trackedEntityId: string;
+    };
+  };
+  country_code?: string | null;
 };
 
 export type DataElementDataService = {
   data_element_code: string;
   country_code: string;
   service_type: ServiceType;
-  service_config: Record<string, any>;
+  service_config: {
+    dhisInstanceCode: string;
+  };
 };
 
 export type SupersetInstance = {
   code: string;
-  config: Record<string, any>;
+  config: {
+    serverName: string;
+    baseUrl: string;
+    insecure?: boolean;
+  };
 };
 
 export type DhisInstance = {
   code: string;
   readonly: boolean;
-  config: Record<string, any>;
+  config: {
+    productionUrl: string;
+    devUrl: string;
+  };
 };
 
 export type EntityHierarchy = {
   name: string;
   canonical_types: string[];
 };
 
-type DataElementInstance = DatabaseType<DataElement, BaseDataElementType>;
-export type EntityInstance = DatabaseType<Entity, BaseEntityType>;
+export type DataSourceTypeInstance = {
+  code: string;
+  service_type: ServiceType;
+  config: Record<string, DbValue>;
+  databaseType:
+    | typeof TYPES.DATA_ELEMENT
+    | typeof TYPES.DATA_GROUP
+    | typeof TYPES.DATA_SERVICE_SYNC_GROUP;
+};
+type DataElementType = DatabaseType<DataElement, BaseDataElementType>;
+export type EntityType = DatabaseType<Entity, BaseEntityType>;
```

Comment:
Used `EntityType` instead of `EntityInstance` for the following reasons:

* There are some new `...Instance` db tables, avoid confusion with them
* Consistent with `indicators` types
* Consistent with the base type used (`DatabaseType`)

I think  it's not ideal since someone may think that this is `entity.type` (an enum), but I think it's the best compromise for now.